File: Algorithms/1189. Maximum Number of Balloons/javascript/solution.js
/**
 * @param {string} text
 * @return {number}
 */
var maxNumberOfBalloons = function (text) {
    const counter = {
        b: 0,
        a: 0,
        l: 0,
        o: 0,
        n: 0,
    };
    const doubleChar = new Set(['l', 'o']);
    for (let char of text) {
        if (char in counter) {
            counter[char] += doubleChar.has(char) ? 1 : 2;
        }
    }

    return Math.floor(Math.min(...Object.values(counter)) / 2);
};

File: Algorithms/1189. Maximum Number of Balloons/kotlin/solution.kt
class Solution {
    fun maxNumberOfBalloons(text: String): Int {
        val map = hashMapOf<Char, Int>()
        for (ch in text) {
            map[ch] = (map[ch] ?: 0) + 1
        }

        var cnt = if (map.contains('b')) map['b']!! else 0
        for (ch in "balon") {
            if (ch == 'l' || ch == 'o') {
                cnt = Math.min(cnt, map[ch]!! / 2)
            } else {
                cnt = Math.min(cnt, map[ch]!!)
            }
        }
        return cnt
    }
}
